theme/theme.go

```diff
@@ -61,6 +61,39 @@ func DarkTheme() fyne.Theme {
 	return theme
 }
 
+// shade will darken a light color and lighten a light color by the given % value
+// should use 4,8, 12 or 14 percent darken/lighten to implement google material design rules
+func Shade(c color.Color, pct uint32) color.Color {
+	r, g, b, a := c.RGBA()
+	if pct == 0 {
+		return color.NRGBA{uint8(r), uint8(g), uint8(b), uint8(a)}
+	}
+	if pct > 50 {
+		pct = 50
+	}
+	if r+g+b < 3*0x8080 {
```

Comment:
it seems like this might be important, should it be extracted into something like an `isLight` helper?